cmd/web-handlers.go

```diff
@@ -546,10 +546,15 @@ func (web *webAPIHandlers) Download(w http.ResponseWriter, r *http.Request) {
 	objectLock.RLock()
 	defer objectLock.RUnlock()
 
-	if err := objectAPI.GetObject(bucket, object, 0, -1, w); err != nil {
+	info, err := objectAPI.GetObjectInfo(bucket, object)
```

Comment:
After discussion with @harshavardhana - it will be better to modify in https://github.com/minio/minio/blob/master/cmd/gateway-s3.go#L262 - it avoids a network call to the backend gateway. We can make the change specifically for s3 gateway